### PR TITLE
Add dem-stitcher and enforce versioning for scipy and python

### PR DESCRIPTION
**Pull request by sssangha**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] ~~Reset the build number to `0` (if the version changed)~~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by sssangha** [comment]
Addresses issue #4

**Comment by cmarshak** [comment]
I have only made changes to the recipe *after* I have a release of of the software. But that's due to my limited experience. @jhkennedy might have more nuanced view.

Based on the checklist in the automated PR description, the`number` under the `build` group was incremented because the software was not re-released. Learned something new!

**Review by cmarshak — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ requirements:
     - {{ compiler('cxx') }}
     - cmake
   host:
-    - python
+    - python >=3.8
```
Comment (jhkennedy) [inline recipe/meta.yaml:33]:
```suggestion
    - python
```
The Python version constraint cannot be listed here for non-noarch packaged and instead is controlled by the build section; see line 18.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -54,9 +54,9 @@ requirements:
     - pip
     - pyproj
     - requests
-    - scipy <1.10
-    - mintpy
+    - scipy>1.10.0
```
Comment (jhkennedy) [inline recipe/meta.yaml:57]:
```suggestion
    - scipy >1.10.0
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -41,11 +41,11 @@ requirements:
   run:
     - python
     - asf_search
-    - cartopy
+    - dem_stitcher >=2.5.0
     - gdal >=3.4.1
     - h5py
     - joblib
-    - matplotlib-base
+    - matplotlib
```
Comment (jhkennedy) [inline recipe/meta.yaml:48]:
```suggestion
    - matplotlib-base
```

**Review by jhkennedy — COMMENTED** [review]
Adding code suggestions for the linting errors in https://github.com/conda-forge/aria-tools-feedstock/pull/6#issuecomment-2057515575

File: recipe/meta.yaml (hunk 3 above)
Comment (jhkennedy) [reply recipe/meta.yaml:48]: Note: ARIA-tools does not depend on qt, so we do want matplotlib-base here.

**Comment by jhkennedy** [comment]
@sssangha @cmarshak, this recipe is now passing the linter and tests, but I'm still not sure we've got the dependencies right. 

Notably, this recipe is for v1.1.6, and when the recipe was updated for this version, the CI/CD tooling highlighted a number of dependency issues that appear to have been ignored; see the "Dependency Analysis" section here: https://github.com/conda-forge/aria-tools-feedstock/pull/2#issue-1844216331


The dependency analysis caught these issues, which are reflected in this PR:
```diff
- cartopy
- mintpy
+ dem_sticher
```

But, this PR **does not** reflect these other issues that were caught
```diff
- h5py
- xarray
+ numpy
+ arm_pyart
+ pulp
+ python-dateutil
 ```

And this PR also does this, which was not flagged by the dependency analysis:
```diff
+ dask
```

Can you review each of these for v1.1.16?

**Comment by jhkennedy** [comment]
@conda-forge-admin, please rerender

**Pull request merged**